kernel/src/mem.rs

```diff
@@ -91,7 +91,9 @@ impl<L, T> AppSlice<L, T> {
                 .process
                 .kernel
                 .process_map_or(false, appid.idx(), |process| {
-                    process.add_mpu_region(self.ptr() as *const u8, self.len() as u32)
+                    process
+                        .add_mpu_region(self.ptr() as *const u8, self.len(), self.len())
```

Comment:
looks suspicious by passing the same argument twice

Reply:
These arguments represent the maximum and minimum region size, correspondingly. Considering that for IPC we want to pass in the exact region start and end address, allowing for no flexibility in region placement, in the case of IPC these end up both being `self.len()`.